File: 94. Binary Tree Inorder Traversal.cpp
class Solution {
public:
    void solve(TreeNode *root,vector<int> &result)
    {
        if(root == NULL){
            return ;
        }
        solve(root->left,result);
        result.push_back(root->val);
        solve(root->right,result);
    }
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> result;
        solve(root,result);
        return result;
    }
};

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> list=new ArrayList<>();
        inOrder(root,list);
        return list;
    }
    void inOrder(TreeNode root,List<Integer> list){
        if(root==null)return;
        inOrder(root.left,list);
        list.add(root.val);
        inOrder(root.right,list);
    }
}